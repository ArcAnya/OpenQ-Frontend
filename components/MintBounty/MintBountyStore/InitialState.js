const InitialState = {
	bounty: null,
	bountyAddress: '',
	bountyExists: false,
	walletConnected: false,
	enableMint: false,
	error: null,
	errorMessage: '',
	isBountyMinted: false,
	isLoading: false,
	isValidUrl: false,
	issueClosed: false,
	issueData: '',
	issueFound: false,
	issueId: '',
	issueNumber: '',
	issueUrl: '',
	orgId: '',
	repoName: '',
	transactionPending: false
};

export default InitialState;