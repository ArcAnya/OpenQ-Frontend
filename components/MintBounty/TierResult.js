import React, { useRef } from 'react';

const TierResult = ({ finalTierVolumes, sum }) => {
  const container = useRef();
  return (
    <>
      Total
      <div className={`flex w-11/12 text-sm  w-full items-center gap-2 mb-1`}>
        <div className='w-9 flex-none'>{sum}%</div>
        <div
          ref={container}
          className={`flex border-transparent border-2 flex-none rounded-full overflow-hidden w-full relative ${
            sum < 100 ? 'border-gray-500' : sum > 100 ? 'border-red-500' : 'border-primary'
          }`}
        >
          {finalTierVolumes.map((volume, index) => {
            const saturation = index % 2 ? 84 - index : 84 - index + 1;
            const lightness = !(index % 2) ? 48 + index : 48 + index - 1;
            const hue = 400 - index * 67;
            return (
              <div
                key={index}
                style={{
                  width: `${
                    (container.current && container.current.clientWidth * finalTierVolumes[index] + 50) / 100
                  }px`,
                  backgroundColor: `hsl(${hue}, ${saturation}%, ${lightness}%)`,
                }}
                type='textarea'
                className='resize-x h-4 w-full inline z-15 left-0 top-0'
              ></div>
            );
          })}
        </div>
      </div>{' '}
    </>
  );
};
export default TierResult;
