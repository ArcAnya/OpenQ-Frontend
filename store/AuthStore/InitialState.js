const InitialState = {
    isAuthenticated: false,
    user: null,
    token: null,
};

export default InitialState;