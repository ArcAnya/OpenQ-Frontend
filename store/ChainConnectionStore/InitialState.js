const InitialState = {
    hasMetamask: false,
    activeAccount: null,
    showWalletConnect: true,
    fullyConnected: false,
    isOnCorrectNetwork: false,
};

export default InitialState;