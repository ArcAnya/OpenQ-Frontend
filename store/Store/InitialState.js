import Utils from '../../services/utils/Utils';

import OpenQClient from '../../services/ethers/OpenQClient';
import OpenQSubgraphClient from '../../services/subgraph/OpenQSubgraphClient';
import GithubRepository from '../../services/github/GithubRepository';
import TokenClient from '../../services/coins/TokenClient';
import GithubBot from '../../services/github-bot/GithubBot';
import Logger from '../../services/logger/Logger';

import MockGithubRepository from '../../services/github/MockGithubRepository';
import MockOpenQClient from '../../services/ethers/MockOpenQClient';
import MockOpenQSubgraphClient from '../../services/subgraph/MockOpenQSubgraphClient';
import MockTokenClient from '../../services/coins/MockTokenClient';
import MockGithubBot from '../../services/github-bot/MockGithubBot';

// Token Metadata
// Array of all supported tokens
import polygonMainnetTokens from '../../constants/polygon-mainnet-tokens.json';
import mumbaiTokens from '../../constants/polygon-mumbai-tokens.json';
import localTokens from '../../constants/local-tokens.json';

// Mapping of tokens with token metadata for token address lookup
import polygonMainnetTokenMetadata from '../../constants/polygon-mainnet.json';
import mumbaiTokenMetadata from '../../constants/polygon-mumbai.json';
import localTokenMetadata from '../../constants/local.json';

let InitialState = {};
switch (process.env.NEXT_PUBLIC_DEPLOY_ENV) {
	case 'local':
		InitialState = {
			tokenMetadata: localTokenMetadata,
			tokens: localTokens,
			openQClient: new MockOpenQClient(),
			githubRepository: new MockGithubRepository(),
			openQSubgraphClient: new MockOpenQSubgraphClient(),
			tokenClient: new MockTokenClient(),
			githubBot: new MockGithubBot(),
			logger: new Logger(true),
			utils: new Utils(),
		};
		break;
	case 'docker':
		InitialState = {
			tokenMetadata: localTokenMetadata,
			tokens: localTokens,
			openQClient: new OpenQClient(),
			githubRepository: new GithubRepository(),
			openQSubgraphClient: new OpenQSubgraphClient(),
			tokenClient: new TokenClient(),
			githubBot: new MockGithubBot(),
			logger: new Logger(true),
			utils: new Utils(),
		};
		break;
	case 'development':
		InitialState = {
			tokenMetadata: mumbaiTokenMetadata,
			tokens: mumbaiTokens,
			openQClient: new OpenQClient(),
			githubRepository: new GithubRepository(),
			openQSubgraphClient: new OpenQSubgraphClient(),
			tokenClient: new TokenClient(),
			githubBot: new MockGithubBot(),
			logger: new Logger(true),
			utils: new Utils(),
		};
		break;
	case 'staging':
		InitialState = {
			tokenMetadata: polygonMainnetTokenMetadata,
			tokens: polygonMainnetTokens,
			openQClient: new OpenQClient(),
			githubRepository: new GithubRepository(),
			openQSubgraphClient: new OpenQSubgraphClient(),
			tokenClient: new TokenClient(),
			githubBot: new GithubBot(),
			logger: new Logger(true),
			utils: new Utils(),
		};
		break;
	case 'production':
		InitialState = {
			tokenMetadata: polygonMainnetTokenMetadata,
			tokens: polygonMainnetTokens,
			openQClient: new OpenQClient(),
			githubRepository: new GithubRepository(),
			openQSubgraphClient: new OpenQSubgraphClient(),
			tokenClient: new TokenClient(),
			githubBot: new GithubBot(),
			logger: new Logger(false),
			utils: new Utils(),
		};
		break;
	default:
		throw Error('ENVIRONMENT NOT CONFIGURED CORRECTLY. Set an environment with DEPLOY_ENV');
}

export default InitialState;