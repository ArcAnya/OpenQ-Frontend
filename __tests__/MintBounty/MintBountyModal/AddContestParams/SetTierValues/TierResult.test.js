/**
 * @jest-environment jsdom
 */
import React from 'react';
import TierResult from '../../../../../components/MintBounty/MintBountyModal/AddContestParams/SetTierValues/TierResult';
import InitialState from '../../../../../store/Store/InitialState';
import renderer from 'react-test-renderer';

InitialState.openQClient.shouldSleep = 200;

describe('Tier Tier Result', () => {
  it('should match DOM Snapshot', () => {
    const tree = renderer.create(<TierResult sum={100} finalTierVolumes={[20, 80]} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
