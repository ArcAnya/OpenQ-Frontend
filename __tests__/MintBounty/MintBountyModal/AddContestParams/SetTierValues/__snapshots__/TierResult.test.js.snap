// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tier Tier Result should match DOM Snapshot 1`] = `
[
  <div
    className="flex items-center gap-2 font-semibold"
  >
    Total
  </div>,
  <div
    className="flex w-11/12 text-sm  items-center gap-2 mb-1 pl-2 pr-8"
  >
    <div
      className="w-9 flex-none"
    >
      100
      %
    </div>
    <div
      className="flex border-transparent border-2 flex-none rounded-full overflow-hidden w-full relative border-primary"
    >
      <div
        className="resize-x h-4 w-full inline z-15 left-0 top-0"
        style={
          {
            "backgroundColor": "hsl(400, 85%, 48%)",
            "width": "NaNpx",
          }
        }
        type="textarea"
      />
      <div
        className="resize-x h-4 w-full inline z-15 left-0 top-0"
        style={
          {
            "backgroundColor": "hsl(333, 83%, 48%)",
            "width": "NaNpx",
          }
        }
        type="textarea"
      />
    </div>
  </div>,
  " ",
]
`;
